src/core/assertions.js

```diff
@@ -92,36 +97,57 @@ exports.assertObjectMatchSpec = (object, spec, exact = false) => {
         const currentValue = _.get(object, field)
         const expectedValue = Cast.value(value)
 
-        switch (matcher) {
-            case 'match':
-            case 'matches':
-                expect(
-                    currentValue,
-                    `Property '${field}' (${currentValue}) does not match '${expectedValue}'`
-                ).to.match(new RegExp(expectedValue))
-                break
+        if (matcher.indexOf('matches') !== -1 || matcher.indexOf('match') !== -1) {
+            expect(
+                currentValue,
+                `Property '${field}' (${currentValue}) does not match '${expectedValue}'`
+            ).to.match(new RegExp(expectedValue))
+        }
 
-            case 'contain':
-            case 'contains':
-                expect(
-                    currentValue,
-                    `Property '${field}' (${currentValue}) does not contain '${expectedValue}'`
-                ).to.contain(expectedValue)
-                break
+        if (matcher.indexOf('contain') !== -1 || matcher.indexOf('contains') !== -1) {
+            expect(
+                currentValue,
+                `Property '${field}' (${currentValue}) does not contain '${expectedValue}'`
+            ).to.contain(expectedValue)
+        }
+
+        if (matcher.indexOf('defined') !== -1 || matcher.indexOf('present') !== -1) {
+            expect(currentValue, `Property '${field}' is undefined`).to.not.be.undefined
+        }
 
-            case 'defined':
-            case 'present':
-                expect(currentValue, `Property '${field}' is undefined`).to.not.be.undefined
-                break
+        const dateOffsetRegexMatch = dateOffsetRegex.exec(matcher)
+        if (dateOffsetRegexMatch !== null) {
+            const [, amount, , unit, locale] = dateOffsetRegexMatch
+            try {
+                const localeNormalize = Intl.getCanonicalLocales(locale)[0]
+                const expectedDate = moment()
+                    .add(amount, unit)
+                    .locale(localeNormalize)
+                    .format(expectedValue)
 
-            case 'equal':
-            case 'equals':
-            default:
                 expect(
                     currentValue,
-                    `Expected property '${field}' to equal '${value}', but found '${currentValue}'`
-                ).to.be.deep.equal(expectedValue)
+                    `Expected property '${field}' to equal '${expectedDate}', but found '${currentValue}'`
+                ).to.be.deep.equal(expectedDate)
+            } catch (e) {
+                if (e instanceof RangeError) {
+                    throw new Error(`invalid locale of ${locale}`)
```

Comment:
Not sure we need a specific message here, we could get rid of the catch and let it throw :)
`expected output: RangeError: invalid language tag: EN_US` seems clear enough, what do you think?

Reply:
I agree with tou, it's even clearer to me